### DISCUSSION TITLE
Understanding data parallelism in DeePMD

### DISCUSSION THREAD

**Discussion by matr1x-1** [deepmodeling/deepmd-kit#3341] [Q&A] [answered]
I have a question about using MPI. Specifically, the documentation says: "Note that the parallelism mode is data parallelism, so it is not expected to see the training time per batch decreases." When I run a job (with "--mpi-log=workers") on 2 GPUs, I get something like this:

> DEEPMD rank:1  INFO    batch    8100 training time 8.88 s, testing time 0.00 s, total wall time 8.90 s
> DEEPMD rank:0  INFO    batch    8100 training time 8.70 s, testing time 0.05 s, total wall time 8.90 s

How do I have to interpret this? As stated above, the training time stays the same with 1 or 2 GPUs. The batch, here 8100, also appears twice. Is something set up incorrectly in this case as I don't want to calculate the same batch twice or is this the expected behaviour?
Thanks for your help!

**Comment by njzjz** [comment, marked as answer]
Two ranks are initialized using different random seeds, so their training data in each batch is different.

**Reply by matr1x-1** [reply]
Ok thanks, but doesn't this mean that when I set e.g. numb_steps=10000 and use 2 GPUs then it trains it on 20000 batches?

**Reply by njzjz** [reply]
Yes, it's the expected behavior.

**Reply by matr1x-1** [reply]
I cannot set the seed myself, right?
